corehq/apps/es/utils.py

```diff
@@ -48,10 +49,15 @@ def track_es_report_load(domain, report_slug, owner_count):
         )
 
 
-def es_format_datetime(date_or_datetime):
+def es_format_datetime(val):
     """
     Takes a date or datetime object and converts it to a format ES can read
-    (see DATE_FORMATS_ARR). Timezone-aware values are converted to UTC. Strings
-    are returned unmodified.
+    (see DATE_FORMATS_ARR). Strings are returned unmodified.
     """
-    return date_or_datetime if isinstance(date_or_datetime, str) else date_or_datetime.isoformat()
+    if isinstance(val, str):
+        return val
+    elif isinstance(val, datetime) and val.microsecond and val.tzinfo:
+        # We don't support microsec precision with timezones
+        return val.astimezone(timezone.utc).replace(tzinfo=None).isoformat()
+    else:
+        return val.isoformat()
```

Comment:
I opted to only change behavior for datetimes with microseconds and timezones, leaving everything else alone.  This way we can be pretty confident in the safety of this change, as the only change is to a scenario which currently errors.

Reply:
I'm guessing you chose to keep the microseconds and drop the timezone, instead of the other way round, because the microseconds are more useful in the context of ES for sorting. Is that right? Or is it safe to assume that all datetimes without timezones in ES are UTC, and so after converting to UTC the timezone is unnecessary?

Reply:
Yeah, microseconds are essential for pagination, as we really need a unique value.  Actually that reminds me, I meant to comment on the decision to use `gt` instead of `gte`.  But yes, generally I'm assuming that we store all dates in UTC.  If we did store a tz aware datetime in ES, I _think_ ES would convert it to UTC anyways, so querying would behave sensibly.  The stored value and this query value are interpreted and compared intelligently by ES. I'm not sure how they handle the case where one has a timezone and the other doesn't, but I'd guess they'd assume no timezone means UTC.

Aside: I've been thinking of "server time" as being UTC equivalent but without an explicit timezone or offset specified (timezone-naive).  I believe that's how we conceptualize it, as per:
https://github.com/dimagi/commcare-hq/blob/374198cce40042cfeb57855b2b101e8b50447231/corehq/util/timezones/conversions.py#L53-L55